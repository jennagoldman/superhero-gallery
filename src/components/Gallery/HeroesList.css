.heroesList {
  list-style: none;
  text-align: center;
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
}

.heroListItem {
  display: inline-block;
  margin: 2rem;
}

