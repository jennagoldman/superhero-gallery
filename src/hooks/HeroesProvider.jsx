import React, { createContext, useState, useEffect, useContext } from 'react';
import { fetchHeroes } from '../services/marvel-api.js';
import PropTypes from 'prop-types';

export const HeroesContext = createContext();

export const HeroesProvider = ({ children }) => {
  const [heroes, setHeroes] = useState([]);
  const [offset, setOffset] = useState(0);

  useEffect(() => {
    fetchHeroes(0)
      .then(heroes => setHeroes(heroes));
  }, []);

  useEffect(() => {
    if(offset > 0) {
      fetchHeroes(offset)
        .then(heroes => setHeroes(heroes));
    }
  }, [offset]);

  const handleOffset = by => setOffset(prevOffset => prevOffset + by);

  return (
    <HeroesContext.Provider value={{ heroes, offset, handleOffset }}>
      {children}
    </HeroesContext.Provider>
  );
};

HeroesProvider.propTypes = {
  children: PropTypes.node
};

export const useHeroes = () => {
  const { heroes } = useContext(HeroesContext);
  return heroes;
};

export const useOffset = () => {
  const { offset } = useContext(HeroesContext);
  return offset;
};

export const useHandleOffset = () => {
  const { handleOffset } = useContext(HeroesContext);
  return handleOffset;
};
