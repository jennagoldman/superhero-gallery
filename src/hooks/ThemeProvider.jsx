import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

export const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const toggleTheme = ({ target }) => {
    if(target.checked) setTheme('dark');
    if(!target.checked) setTheme('light');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node
};

export const useTheme = () => {
  const { theme } = useContext(ThemeContext);
  return theme;
};

export const useToggle = () => {
  const { toggleTheme } = useContext(ThemeContext);
  return toggleTheme;
};
