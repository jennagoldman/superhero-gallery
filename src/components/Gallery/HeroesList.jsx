import React from 'react';
import { useHeroes } from '../../hooks/HeroesProvider.jsx';
import styles from './HeroesList.css';

const HeroesList = () => {
  const heroes = useHeroes();

  const heroNodes = heroes.map(hero => (
    <li key={hero.id}>
      <p>{hero.name}</p>
      <img src={hero.imageUrl} alt={hero.name} />
      <p>{hero.description}</p>
      <a href={hero.comicsUrl}></a>
    </li>
  ));
  return (
    <ul>
      {heroNodes}
    </ul>
  );
};

export default HeroesList;
