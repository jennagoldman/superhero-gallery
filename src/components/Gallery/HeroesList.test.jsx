import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from '../../hooks/ThemeProvider.jsx';
import { HeroesProvider } from '../../hooks/HeroesProvider.jsx';
import HeroesList from './HeroesList.jsx';

describe('Gallery', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <ThemeProvider>
        <HeroesProvider>
          <HeroesList />
        </HeroesProvider>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
