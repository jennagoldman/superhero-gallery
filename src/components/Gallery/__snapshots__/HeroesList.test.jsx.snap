// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery matches a snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "theme": "light",
      "toggleTheme": [Function],
    }
  }
>
  <HeroesProvider>
    <HeroesList />
  </HeroesProvider>
</ContextProvider>
`;
